src/crate/client/http.py

```diff
@@ -159,6 +159,12 @@ def _get_sql_path(self, **url_params):
     def _update_server_pool(self, servers, **kwargs):
         for server in servers:
             if not server in self.server_pool:
+                https = server.lower().startswith('https:')
+                if not https:
+                    #clean up any kwargs which are invalid for HTTPConnectionPool
+                    for key in ['ca_certs', 'cert_reqs']:
+                        if key in kwargs:
+                            del kwargs[key]
```

Comment:
I think this still won't work if there are https and non-https server in the list since the kwargs is shared.
You'd have to copy it once for http and once for https.